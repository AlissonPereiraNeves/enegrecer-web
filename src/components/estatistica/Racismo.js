import React, { Component } from 'react';
import firebaseApp from '../../utils/firebaseUtils';
import TipoDeEstatistica from './TipoDeEstatistica';

export default class Racismo extends Component {
  constructor(props) {
    super(props);
    this.state = { totalDeDenunciaPorRacismo: '0' };
  }

  componentDidMount() {
    const ref = firebaseApp.database().ref();
    ref.child('denuncias').orderByChild('idCategoria').equalTo('racismo')
      .once('value', (snapshot) => {
        this.setState({ totalDeDenunciaPorRacismo: snapshot.numChildren() });
      });
  }

  render() {
    return (<TipoDeEstatistica
      id="vermelho"
      numero={this.state.totalDeDenunciaPorRacismo}
      categoria="Racismo"
    />);
  }
}
