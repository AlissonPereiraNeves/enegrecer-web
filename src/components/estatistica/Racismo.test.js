import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Racismo from './Racismo';

describe('Racismo', () => {
  it('deve construir a pagina corretamente ', () => {
    const racismo = shallow(<Racismo id="vermelho" />);
    const racismoJson = toJson(racismo);
    expect(racismoJson).toMatchSnapshot();
  });
});
