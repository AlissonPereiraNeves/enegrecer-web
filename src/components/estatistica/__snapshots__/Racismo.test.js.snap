// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Racismo deve construir a pagina corretamente  1`] = `
<TipoDeEstatistica
  categoria="Racismo"
  id="vermelho"
  numero="0"
/>
`;
